draft-ietf-jsonpath-base.md

```diff
@@ -1456,19 +1459,19 @@ The type correctness of function expressions can now be defined in terms of this
 
 A function expression is correctly typed if all the following are true:
 
-* If it occurs as a `filter-path` in a test expression, the function
-is defined to have result type `OptionalNodesType` or one of its subtypes,
-or to have result type `OptionalBooleanType` or one of its subtypes.
+* If it occurs directly in a test expression, the function
+is defined to have a result type out of `ST(OptionalNodesType)`.
+or to have a result type out of `ST(OptionalBooleanType)`.
```

Comment:
I think "out" is redundant here.  "Result" implies output.

Reply:
"out of" as in "two out of three".  Do you have a better way to say this?

Reply:
I see what you're getting at now.  I would say "defined by".

> the function has a result type defined by `ST(OptionalNodesType)`

with `ST(T)` being the set of the type T and all of its subtypes (as you've defined).

Reply:
Hmm, "take a lolly defined by this jar" doesn't work for me, and I think most people wouldn't think that a set "defines" its members.  But maybe "in" is better than "out of"...